TykTechnologies/tyk-operator
# Allow subgraphs' header field to take null values

## Description
While exporting GraphQL ApiDefinitions in older versions of Tyk via snapshot tool, `spec.graphql.supergraph.subgraphs[].headers` field is encoded as `null` because in Tyk v4.0, this field is not introduced and recognized as null value. Once users try to create the exported manifest, validation on Kubernetes will fail since `headers` cannot take null values. 



## Related Issue




## Motivation and Context


## Test Coverage For This Change



- Create GraphQL APIs on v4.0,
- Export it via Snapshot tool,
- Apply it,
- Kubernetes shouldn't throw any validation error and Operator should create the resource on Tyk.

It needs testing for v4.0 and newer ones (v4.3 or v5.0)

## Screenshots (if appropriate)

## Types of changes


## Checklist
